hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/VertexAPI.java

```diff
@@ -259,22 +307,61 @@ private static void checkBatchSize(HugeConfig config,
         int max = config.get(ServerOptions.MAX_VERTICES_PER_BATCH);
         if (vertices.size() > max) {
             throw new IllegalArgumentException(String.format(
-                      "Too many counts of vertices for one time post, " +
+                      "Too many vertices for one time post, " +
                       "the maximum number is '%s'", max));
         }
     }
 
-    @JsonIgnoreProperties(value = {"type"})
-    private static class JsonVertex implements Checkable {
+    private Id getVertexId(HugeGraph g, JsonVertex vertex) {
+        VertexLabel vertexLabel = g.vertexLabel(vertex.label);
+        String labelId = vertexLabel.id().asString();
+        IdStrategy idStrategy = vertexLabel.idStrategy();
+        E.checkArgument(idStrategy != IdStrategy.AUTOMATIC,
+                        "Automatic Id strategy is not supported now");
+
+        if (idStrategy == IdStrategy.PRIMARY_KEY) {
+            List<Id> pkIds = vertexLabel.primaryKeys();
+            List<Object> pkValues = new ArrayList<>(pkIds.size());
+            for (Id pkId : pkIds) {
+                String propertyKey = g.propertyKey(pkId).name();
+                Object propertyValue = vertex.properties.get(propertyKey);
+                E.checkState(propertyValue != null,
+                             "The value of primary key '%s' can't be null",
+                             propertyKey);
+                pkValues.add(propertyValue);
+            }
+
+            String value = SplicingIdGenerator.concatValues(pkValues);
+            return SplicingIdGenerator.splicing(labelId, value);
+        } else {
+            assert idStrategy == IdStrategy.CUSTOMIZE_NUMBER ||
+                   idStrategy == IdStrategy.CUSTOMIZE_STRING;
+            return HugeVertex.getIdValue(vertex.id);
+        }
+    }
+
+    private static class VertexRequest {
+
+        @JsonProperty("vertices")
+        public List<JsonVertex> jsonVertices;
+        @JsonProperty("update_strategies")
+        public Map<String, UpdateStrategy> updateStrategies;
+        @JsonProperty("create_if_not_exist")
+        public boolean createIfNotExist = true;
+
+        @Override
+        public String toString() {
+            return String.format("VertexRequest{jsonVertices=%s," +
+                                 "updateStrategies=%s,createIfNotExist=%s}",
+                                 this.jsonVertices, this.updateStrategies,
+                                 this.createIfNotExist);
+        }
+    }
+
+    private static class JsonVertex extends JsonElement {
```

Comment:
override equals() & hashCode() method.
```
equals(JsonEdge other): return this.id.equals(other.id)
hashCode(): return this.id.hashCode()
```
Note: make sure that update JsonVertex.id in getVertexId() method.

The same as JsonEdge class, maybe it's better to move id field to JsonElement class.